llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp

```diff
@@ -12,6 +12,8 @@
 // restore correct SPIR-V types of function arguments and return values.
 // This pass also substitutes some llvm intrinsic calls with calls to newly
 // generated functions (as the Khronos LLVM/SPIR-V Translator does).
+// This pass also trims convergence intrinsics as those were only useful when
+// modifying the CFG during IR passes.
 //
 // NOTE: this pass is a module-level one due to the necessity to modify
```

Comment:
Should we consider putting this new behavior in a `FunctionPass` rather than the existing `ModulePass` since it doesn't have any module-level behavior? From the documentation it looks like that's preferred when possible: https://llvm.org/docs/WritingAnLLVMPass.html

Reply:
That's fine for me, will rebase and move this logic into a new pass.

Reply:
Done, forced pushed for rebase and since this means mostly a rewrite